satpy/dataset.py

```diff
@@ -316,10 +666,32 @@ def create_filtered_dsid(dataset_key, **dfilter):
             ds_dict = {'name': dataset_key}
         elif isinstance(dataset_key, numbers.Number):
             ds_dict = {'wavelength': dataset_key}
-    for key, value in dfilter.items():
-        if value is not None:
-            ds_dict.setdefault(key, value)
-    return DatasetID.from_dict(ds_dict)
+        else:
+            raise TypeError("Don't know how to interpret a dataset_key of type {}".format(type(dataset_key)))
+    if filter_query is not None:
+        for key, value in filter_query._dict.items():
+            if value != '*':
+                ds_dict.setdefault(key, value)
+
+    return DataQuery.from_dict(ds_dict)
+
+
+def create_filtered_id(dataset_key, filter_query):
+    """Create a DatasetID matching *dataset_key* and *filter_query*.
+
+    If a proprety is specified in both *dataset_key* and *filter_query*, the former
```

Comment:
`proprety` -> `property`

Reply:
I'm removing this function as it isn't used anywhere, in favour of `create_filtered_query`